cmake/onnxruntime_fuzz_test.cmake

```diff
@@ -5,61 +5,81 @@
 # the fuzzing project
 if (onnxruntime_FUZZ_ENABLED)
 	message(STATUS "Building dependency protobuf-mutator and libfuzzer")
-	
+
 	# set the options used to control the protobuf-mutator build
 	set(PROTOBUF_LIBRARIES ${PROTOBUF_LIB})
 	set(LIB_PROTO_MUTATOR_TESTING OFF)
-	
+
 	# include the protobuf-mutator CMakeLists.txt rather than the projects CMakeLists.txt to avoid target clashes
 	# with google test
 	add_subdirectory("external/libprotobuf-mutator/src")
-	
+
 	# add the appropriate include directory and compilation flags
 	# needed by the protobuf-mutator target and the libfuzzer
 	set(PROTOBUF_MUT_INCLUDE_DIRS "external/libprotobuf-mutator")
 	onnxruntime_add_include_to_target(protobuf-mutator ${PROTOBUF_LIB})
 	onnxruntime_add_include_to_target(protobuf-mutator-libfuzzer ${PROTOBUF_LIB})
 	target_include_directories(protobuf-mutator PRIVATE ${INCLUDE_DIRECTORIES} ${PROTOBUF_MUT_INCLUDE_DIRS})
 	target_include_directories(protobuf-mutator-libfuzzer PRIVATE ${INCLUDE_DIRECTORIES} ${PROTOBUF_MUT_INCLUDE_DIRS})
-	target_compile_options(protobuf-mutator PRIVATE "/wd4244" "/wd4245" "/wd4267" "/wd4100" "/wd4456")
-	target_compile_options(protobuf-mutator-libfuzzer PRIVATE "/wd4146" "/wd4267")
-	
-	# add Fuzzing Engine Build Configuration 
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        # MSVC-specific compiler options
+        target_compile_options(protobuf-mutator PRIVATE "/wd4244" "/wd4245" "/wd4267" "/wd4100" "/wd4456")
```

Comment:
Usually we should not disable "4244" warnings. But, we don't distribute this component, thus we can make an exception for this.